documentation/course/e2/README.md

```diff
@@ -385,44 +386,52 @@ TODO @piterek130
     <th>Wady</th>
   </tr>
   <tr>
-    <th>Rozwiązanie 1</th>
+    <th>Wzorzec API Gateway</th>
     <td>
       <ul>
-        <li>Zaleta 1</li>
+        <li>Centralizacja autoryzacji i uwierzytelniania</li>
+        <li>Większe bezpieczeństwo aplikacji</li>
+        <li>Ułatwiony routing żądań</li>
+        <li>Wysoka elastyczność</li>
       </ul>
     </td>
     <td>
       <ul>
-        <li>Wada 1</li>
+        <li>Dodatkowa warstwa do wdrożenia i utrzymania</li>
+        <li>Dodatkowy pojedynczy punkt awarii (Single Point of Failure)</li>
       </ul>
     </td>
   </tr>
   <tr>
-    <th>Rozwiązanie 2</th>
+    <th>Bez API Gateway</th>
     <td>
       <ul>
-        <li>Zaleta 1</li>
+        <li>Prostsza implementacja w początkowej fazie projektu</li>
+        <li>Brak dodatkowej warstwy pośredniej</li>
       </ul>
     </td>
     <td>
       <ul>
-        <li>Wada 1</li>
+        <li>Wysoka złożoność dla frontendu</li>
+        <li>Problemy z zabezpieczeniami</li>
+        <li>Klient musi znać adresy każdego serwisu</li>
       </ul>
     </td>
   </tr>
 </table>
 
-**Decyzja:**
+**Decyzja:** Zdecydowano się na zastosowanie API Gateway jako centralnego punktu wejściowego do systemu. API Gateway pełni funkcję pośrednika między klientami a mikroserwisami backendowymi, który przetwarza i kieruje żądania do odpowiednich usług. Rozwiązanie to zwiększa bezpieczeństwo systemu, eliminując potrzebę bezpośredniej komunikacji klientów z wieloma mikroserwisami. API Gateway umożliwia także buforowanie odpowiedzi, ograniczanie liczby żądań i optymalizowanie komunikacji między klientami, a mikroserwisami co pozwala na uzyskanie lepszej wydajności systemu.
```

Comment:
**nit:** You could make the selected choice bold, i.e. `zastosowanie **API Gateway** jako`, me and Przemek have done so, so we could have some consistency.

Same in other mechanisms.